Backpack/src/main/java/net/skyscanner/backpack/toggle/BpkSwitch.kt

```diff
@@ -20,64 +20,85 @@ package net.skyscanner.backpack.toggle
 
 import android.content.Context
 import android.content.res.ColorStateList
+import android.graphics.Color
 import android.util.AttributeSet
-import androidx.appcompat.widget.SwitchCompat
+import androidx.appcompat.content.res.AppCompatResources
+import androidx.appcompat.view.ContextThemeWrapper
+import com.google.android.material.materialswitch.MaterialSwitch
 import net.skyscanner.backpack.R
+import net.skyscanner.backpack.util.colorStateList
 import net.skyscanner.backpack.util.createContextThemeWrapper
 import net.skyscanner.backpack.util.use
 
-private fun wrapContext(context: Context, attrs: AttributeSet?): Context {
-    val withBaseStyle = createContextThemeWrapper(context, attrs, androidx.appcompat.R.attr.switchStyle)
-    return createContextThemeWrapper(withBaseStyle, attrs, R.attr.bpkSwitchStyle)
-}
-
 /**
  * BpkSwitch allow users to toggle between two states, on or off.
  *
- * This class extends [SwitchCompat] directly and thus follows the same interface and design,
- * with the exception of [SwitchCompat.getTrackTintList] and [SwitchCompat.getThumbTintList] that are set
+ * This class extends [MaterialSwitch] directly and thus follows the same interface and design,
+ * with the exception of [MaterialSwitch.getTrackTintList] and [MaterialSwitch.getThumbTintList] that are set
  * according to Backpack's design.
  *
- * @see SwitchCompat
+ * @see MaterialSwitch
  */
 open class BpkSwitch @JvmOverloads constructor(
     context: Context,
     attrs: AttributeSet? = null,
     defStyleAttr: Int = 0,
-) : SwitchCompat(wrapContext(context, attrs), attrs, defStyleAttr) {
+) : MaterialSwitch(
+    createContextThemeWrapper(
+        ContextThemeWrapper(context, R.style.Widget_Material3_CompoundButton_MaterialSwitch),
+        attrs,
+        R.attr.bpkSwitchStyle,
+    ),
+    attrs,
+    defStyleAttr,
+) {
 
     init {
         initialize(attrs, defStyleAttr)
     }
 
-    fun initialize(attrs: AttributeSet?, defStyleAttr: Int) {
+    private fun initialize(attrs: AttributeSet?, defStyleAttr: Int) {
```

Comment:
This is the breaking change, but I really doubt someone uses this method explicitly